python/calc_refractive_index.py

```diff
@@ -23,99 +23,193 @@
 from __future__ import print_function, division, absolute_import
 import numpy as np
 
```

Comment:
The refraction calculation code is only called a few times now, so I'll leave them as functions. When this code was originally written it was designed to handle arrays for all the parameters to explore general DCR, but most of that functionality has now been removed.